src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+                            if (ObjectAccessor::GetCreature(*me, *itr)->IsWithinDistInMap(me, 10.0))
+                                zombie = ObjectAccessor::GetCreature(*me, *itr);
+                        if (zombie)
+                        {
+                            zombieToBeEatenGUID = zombie->GetGUID(); // save for later use
+                            Talk(EMOTE_SPOTS_ONE);
+
+                            // the soon-to-be-eaten zombie should stop moving and stop attacking
+                            zombie->AI()->SetData(1, STATE_ZOMBIE_TOBE_EATEN); // change the state of the zombie to STATE_ZOMBIE_TOBE_EATEN
+
+                            // gluth should turn toward the zombie and stop AAs on his primary target. He then pauses for a few seconds
+                            me->GetMotionMaster()->MoveFollow(zombie, 0, 0);
+                            me->SetReactState(ReactStates::REACT_PASSIVE);
+                            me->SetFacingToObject(zombie);
+                            me->SetTarget(ObjectGuid::Empty);
+                            state = STATE_GLUTH_EATING;
+
+                            events.ScheduleEvent(EVENT_KILL_ZOMBIE_SINGLE, 2.5 * IN_MILLISECONDS);
+                        }
+
+                        events.ScheduleEvent(EVENT_SEARCH_ZOMBIE_SINGLE, 7 * IN_MILLISECONDS);
                         break;
+                    }
+                    case EVENT_KILL_ZOMBIE_SINGLE:
+                    {
+                        Creature* zombieToBeEaten = ObjectAccessor::GetCreature(*me, zombieToBeEatenGUID);
+                        if (zombieToBeEaten && zombieToBeEaten->IsAlive() && zombieToBeEaten->IsWithinDistInMap(me, 10.0))
+                            DoCast(zombieToBeEaten, SPELL_ZOMBIE_CHOW_SEARCH_SINGLE); // do the killing + healing in done inside by spell script see below
+
+                        zombieToBeEatenGUID = ObjectGuid::Empty;
+                        state = STATE_GLUTH_NORMAL;
+
+                        // and then return on primary target
+                        me->SetReactState(ReactStates::REACT_AGGRESSIVE);
+                        me->SetFacingToObject(nullptr);
+                        me->SetTarget(me->GetVictim()->GetGUID());
+                        me->GetMotionMaster()->MoveChase(me->GetVictim());
+
+                        break;
+                    }
+                    case EVENT_SEARCH_ZOMBIE_MULTI:
+                    {
+                        if (state == STATE_GLUTH_EATING) // skip and simply wait for the next occurence
+                            break;
+
+                        Creature* zombie = nullptr;
+                        for (SummonList::const_iterator itr = summons.begin(); !zombie && itr != summons.end(); ++itr)
+                            if (ObjectAccessor::GetCreature(*me, *itr)->GetExactDist2d(me) <= 18.0)
+                                zombie = ObjectAccessor::GetCreature(*me, *itr);
```

Comment:
i think u should improve this search.

here is an example:

there are 4 zombies, only one if them is in range:
the summonlist could be like this:

zombie1 (not in range), zombie 2 (in range), zombie3 (not in range), zombie 4 (not in range)

like this, for will always iterate along the list till last position. So the value of "Creature* zombie = nullptr;" will only depend on last position, which is not a correct behaviour.

I suggest a while block + a boolean to stop it, in case a valid case is found.